File: Hackerrank/prime_checker.c
import static java.lang.System.in;

class Prime
{
    void checkPrime(int... num)
    {
        int len = num.length;
        Boolean flag = false;
        for(int i = 0; i < len; i += 1)
        {
            if (num[i] == 2 || num[i] == 3)
            {
                System.out.print(num[i]+" ");
            }
            else if (num[i] % 2 != 0 && num[i] > 2)
            {
                for(int j = 3; j <= num[i]/2; j += 1)
                {
                    if (num[i] % j == 0)
                    {
                        flag = true;
                    }
                }
                if (flag == false)
                {
                    System.out.print(num[i]+" ");
                }
                flag = false;
            }
        }
        if(flag == false)
        {
            System.out.println();   
        }
    }
}

File: Hackerrank/Prime_Checker.java
import java.io.*;
import java.util.*;

public class Prime_Checker{

    public static void main(String[] args) {
        /* Enter your code here. Read input from STDIN. Print output to STDOUT. Your class should be named Solution. */
        
        Scanner sc = new Scanner(System.in);
         
       int[] arr = new int[5];
        
        for(int i=0; i<5 ;i++){
          arr[i]=sc.nextInt();  
        }
        
        checkPrime(arr[0]);
        System.out.println();
        checkPrime(arr[0], arr[1]);
        System.out.println();
        checkPrime(arr[0], arr[1], arr[2]);
        System.out.println();
        checkPrime(arr[0], arr[1], arr[2], arr[3], arr[4]);
        
    }
    
    static void checkPrime(int ... arr){
         for(int i=0; i<arr.length; i++){
             boolean isPrime=true;
             if(arr[i]==1){
                 continue;
             }
             for(int j=2;j<arr[i];j++){
                  if(arr[i]%j==0){
                     isPrime=false ;
                  }    
             }
             
             if(isPrime){
                 System.out.print(arr[i]+" ");
             }
             
         }    
    }
}
